lib/fxpay/errors.js

```diff
@@ -0,0 +1,75 @@
+(function() {
+  'use strict';
+
+  var exports = fxpay.errors = {};
+
+  // All error classes will implicitly inherit from this.
+  exports.FxPayError = createError('FxPayError');
+
+
+  exports.AddReceiptError = createError('AddReceiptError');
```

Comment:
I wonder if this could be DRYed up a bit by iterating over strings to create the errors?